pkg/clustermanager/eksa_mover_test.go

```diff
@@ -0,0 +1,126 @@
+package clustermanager_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/golang/mock/gomock"
+	"github.com/onsi/gomega"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/clustermanager/mocks"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+type moverTest struct {
+	gomega.Gomega
+	tb            testing.TB
+	clientFactory *mocks.MockClientFactory
+	ctx           context.Context
+	spec          *cluster.Spec
+	fromClient    kubernetes.Client
+	toClient      kubernetes.Client
+	log           logr.Logger
+	mgmtCluster   *types.Cluster
+	bootstrap     *types.Cluster
+}
+
+func newMoverTest(tb testing.TB) *moverTest {
+	ctrl := gomock.NewController(tb)
+	return &moverTest{
+		tb:            tb,
+		Gomega:        gomega.NewWithT(tb),
+		clientFactory: mocks.NewMockClientFactory(ctrl),
+		ctx:           context.Background(),
+		spec:          test.VSphereClusterSpec(tb, tb.Name()),
+		log:           test.NewNullLogger(),
+		bootstrap: &types.Cluster{
+			KubeconfigFile: "bootstrap-config",
+		},
+		mgmtCluster: &types.Cluster{
+			KubeconfigFile: "my-config",
+		},
+	}
+}
+
+func (a *moverTest) buildClient(err1, err2 error, objs ...kubernetes.Object) {
+	a.fromClient = test.NewFakeKubeClient(clientutil.ObjectsToClientObjects(objs)...)
+	a.toClient = test.NewFakeKubeClient()
+	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.bootstrap.KubeconfigFile).Return(a.fromClient, err1).MaxTimes(1)
+	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.mgmtCluster.KubeconfigFile).Return(a.toClient, err2).MaxTimes(1)
+}
+
+func (a *moverTest) expectClientError() {
+	a.fromClient = test.NewFakeKubeClientAlwaysError()
+	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.bootstrap.KubeconfigFile).Return(a.fromClient, nil)
+	a.toClient = test.NewFakeKubeClientAlwaysError()
+	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.mgmtCluster.KubeconfigFile).Return(a.toClient, nil)
+}
+
+func TestMoverSuccess(t *testing.T) {
+	tt := newMoverTest(t)
+	objs := tt.spec.ClusterAndChildren()
+	tt.buildClient(nil, nil, objs...)
+	m := clustermanager.NewMover(tt.log, tt.clientFactory,
+		clustermanager.WithMoverRetryBackOff(time.Millisecond),
+		clustermanager.WithMoverNoTimeouts(),
+	)
+
+	tt.Expect(m.Move(tt.ctx, tt.spec, tt.bootstrap, tt.mgmtCluster)).To(gomega.Succeed())
```

Comment:
You are missing assertions that check that the objects have actually been created